src/thor/expansion_action.cc

```diff
@@ -0,0 +1,183 @@
+#include "thor/worker.h"
+
+#include "baldr/json.h"
+#include "baldr/rapidjson_utils.h"
+#include "midgard/constants.h"
+#include "midgard/logging.h"
+#include "midgard/polyline2.h"
+#include "midgard/util.h"
+
+using namespace rapidjson;
+using namespace valhalla::midgard;
+
+namespace {
+// these need to represent array types
+enum PropType {
+  kEdgeId,
+  kStatus,
+  kDist,
+  kTime,
+  kCost,
+  // insert new keys before this
+  LAST
+};
```

Comment:
we could make this part of the protobuf request and let people select a repeated list of these to turn on the ones they do and dont want, and protobuf handles that `LAST` value for you too

Reply:
same, will get back in the general section

Reply:
yeah damn it, I'll do that now, sounds actually awesome! I'm really cautious about adding params to the requests, but definitely down for it if you are!